src/stories/2-frontend-engine/on-submit-error/on-submit-error.stories.tsx

```diff
@@ -0,0 +1,135 @@
+						{ label: "Female", value: "female" },
+					],
+				},
+				radio: {
+					uiType: "radio",
+					label: "Excessive Radio Button",
+					options: [
+						{ label: "Apple", value: "Apple" },
+						{ label: "Berry", value: "Berry" },
+						{ label: "Cherry", value: "Cherry" },
+						{ label: "Apple", value: "Apple" },
+						{ label: "Berry", value: "Berry" },
+						{ label: "Cherry", value: "Cherry" },
+						{ label: "Apple", value: "Apple" },
+						{ label: "Berry", value: "Berry" },
+						{ label: "Cherry", value: "Cherry" },
+						{ label: "Apple", value: "Apple" },
+						{ label: "Berry", value: "Berry" },
+						{ label: "Cherry", value: "Cherry" },
+						{ label: "Apple", value: "Apple" },
+						{ label: "Berry", value: "Berry" },
+						{ label: "Cherry", value: "Cherry" },
+						{ label: "Apple", value: "Apple" },
+						{ label: "Berry", value: "Berry" },
+						{ label: "Cherry", value: "Cherry" },
+						{ label: "Apple", value: "Apple" },
+						{ label: "Berry", value: "Berry" },
+						{ label: "Cherry", value: "Cherry" },
+						{ label: "Apple", value: "Apple" },
+						{ label: "Berry", value: "Berry" },
+						{ label: "Cherry", value: "Cherry" },
+						{ label: "Apple", value: "Apple" },
+						{ label: "Berry", value: "Berry" },
+						{ label: "Cherry", value: "Cherry" },
+					],
+				},
+				unit: {
+					label: "Unit Number",
+					uiType: "unit-number-field",
+				},
+				multi: {
+					uiType: "multi-select",
+					label: "Fruits",
+					options: [
+						{ value: "1", label: "1" },
+						{ value: "2", label: "2" },
+						{ value: "3", label: "3" },
+					],
+				},
+				description: {
+					label: "Feedback",
+					uiType: "textarea",
+					rows: 3,
+					resizable: true,
+					validation: [{ required: true }],
+					chipTexts: ["Best", "Good", "Bad", "Horrible"],
+				},
+				...SUBMIT_BUTTON_SCHEMA,
+			},
+		},
+	},
+	validationMode: "onSubmit",
+};
+
+export default {
+	title: "Form/Frontend Engine/On Submit Error",
+	component: null,
+	parameters: {
+		docs: {
+			page: () => (
+				<>
+					<Title>Sample usage for onSubmitError</Title>
+					<Description>
+						onSubmitError prop takes a handler which is invoked when the form encounters validation errors
+						on submission. The callback invocation is delayed to the next tick to ensure the DOM has been
+						updated to allow handling of validation errors via `aria-invalid` attributes. Here is an of
+						example of how onSubmitError can be used.
+					</Description>
+					<Stories includePrimary={true} title="Examples" />
+				</>
+			),
+		},
+	},
+} as Meta;
+
+export const ScrollToError: Story<IFrontendEngineProps> = () => {
+	const ref = useRef<IFrontendEngineRef>();
+
+	return (
+		<>
+			<FrontendEngine
+				data={{ ...DATA }}
+				ref={ref}
+				onSubmitError={(e) => {
+					console.log("Validation Errors: ", e);
+					const invalidElement = document.querySelector("div:has(*[aria-invalid=true]) > label");
```

Comment:
note that support for `:has` still seems to be limited to newer browsers  https://caniuse.com/css-has

Reply:
any suggestions? the intention in the sample was to scroll to the label of the associated input containing validation errors. is there any possibility the aria attribute can be applied to the input (textarea) directly such that we can query directly for the label with the right `for` attribute?

`
	const invalidElement = document.querySelector("*[aria-invalid=true]");
					if(invalidElement) {
						const scrollToElement = document.querySelector(`*[for=${invalidElement.id}]`) || invalidElement;
						scrollToElement.scrollIntoView();
					}
`

Reply:
not ideal, but
1. there's a `data-testid` set on error messages. if no one overrides `data-error-testid` from the schema, then the `data-testid` value is `<field id>-error-message`. get the element with `[data-testid$="error-message]"` and then scroll to its parent node or label sibling
2. just let it not work if it's not a critical feature and the browser user base is small?

Reply:
imo the scrolling behaviour will be up to how an integration service handles so it depends on its use case will be, anyway here are some observations
- there is a need to update the fallback to the selector because `*[for=${invalidElement.id}]` will throw an error if invalidElement has no id (happens to radio buttons)
```
console.log("Validation Errors: ", e);
const invalidElement = document.querySelector("*[aria-invalid=true]");
if (invalidElement && invalidElement.id) {
	document.querySelector(`label[for=${invalidElement.id}]`).scrollIntoView();
} else {
	document.querySelector(`*[aria-invalid=true]`).scrollIntoView();
}
```
- `data-testid` can work unless it defined in the schema indeed, it can work depending on its use case
- unavailability of `:has()` pseudo-element can be overcome with polyfills (disclaimer: I have not tested it)
	- https://www.npmjs.com/package/css-has-pseudo
	- https://github.com/jplhomer/polyfill-css-has

all in all, I think it's not an issue with FEE but rather a concern with integration, i.e. ymmv depending on how you use the library. as a safeguard, we can leave a comment in this story to say the `:has()` pseudo-element is not supported in Firefox yet so use with caution.

Reply:
can we also move this story to the main `frontend-engine.stories.tsx`? as it stands now, it has one story only, we can simplify the hierarchy

Reply:
Updated- 

The example suggested by Wei Li should be generally useful in communicating the intention of the story. I suppose one could just use JS to navigate to specific sibling / parent's sibling but putting that all into examples / comments seems overly prescriptive and I think we can agree is ultimately up to the integrating service's needs?

Reply:
yes the story and its description is sufficient, may I suggest instead of `console.log()`, use storybook's `action()` method instead? it works like this: https://github.com/LifeSG/web-frontend-engine/blob/main/src/stories/2-frontend-engine/frontend-engine.stories.tsx#L245